src/websocket/client.ts

```diff
@@ -76,18 +76,22 @@ export class WsClient {
           }
 
           const msg: any = msgpack.decode(data)
-          if (signalCb && msg.type === 'Signal') {
-            const decodedMessage: SignalResponseGeneric<any> = msgpack.decode(msg.data);
+          if (msg.type === 'Signal') {
+            if (signalCb) {
+              const decodedMessage: SignalResponseGeneric<any> = msgpack.decode(msg.data);
 
-            // Note: holochain currently returns signals as an array of two values: cellId and the seralized signal payload
-            // and this array is nested within the App key within the returned message.
-            const decodedCellId = decodedMessage.App[0];
-            // Note:In order to return readible content to the UI, the signal payload must also be decoded.
-            const decodedPayload = signalTransform(decodedMessage.App[1]);
+              // Note: holochain currently returns signals as an array of two values: cellId and the serialized signal payload
+              // and this array is nested within the App key within the returned message.
+              const decodedCellId = decodedMessage.App[0];
+              // Note:In order to return readible content to the UI, the signal payload must also be decoded.
+              const decodedPayload = signalTransform(decodedMessage.App[1]);
 
-            // Return a uniform format to UI (ie: { type, data } - the same format as with callZome and appInfo...)
-            const signal: AppSignal = { type: msg.type , data: { cellId: decodedCellId, payload: decodedPayload }};
-            signalCb(signal);
+              // Return a uniform format to UI (ie: { type, data } - the same format as with callZome and appInfo...)
+              const signal: AppSignal = { type: msg.type , data: { cellId: decodedCellId, payload: decodedPayload }};
+              signalCb(signal);
+            } else {
+              console.error(`Received signal but no signal callback was set in constructor`);
```

Comment:
should this be a console.error? not setting a signalCb seems like a common use possibility

Reply:
the way I see it is:
- if holochain is sending signals and there's no signalCB then it is an error
So not setting a signalCb in an application that does not use signals is fine and will never log an error

thoughts?

Reply:
Mostly fair point. One more thing to consider is that for a given holochain conductor, you wish (and I have done this) to construct multiple clients at different points and places in the code. Some may attach the callback listener, and provide the one and only handler, and others may not. The one that doesn't will log this error despite you having another one which is handling them

Reply:
Yes, I see. So maybe a better solution would be to `console.log` it once during existence of this class instance? Easy to do and quite elegant I think.

Reply:
That sounds great!

Reply:
Yep!

Reply:
cf5cb68 should do the job